START TIME: Thu May 22 16:09:45 CEST 2025
slurmstepd: error: couldn't chdir to `/workspace': No such file or directory: going to /tmp instead
slurmstepd: error: couldn't chdir to `/workspace': No such file or directory: going to /tmp instead
slurmstepd: error: couldn't chdir to `/workspace': No such file or directory: going to /tmp instead
slurmstepd: error: couldn't chdir to `/workspace': No such file or directory: going to /tmp instead
2025-05-22 16:10:07,423 - root - INFO - Setting seed to 42
2025-05-22 16:10:07,423 - root - INFO - Experiment args: Namespace(dataset='/capstor/store/cscs/ethz/large-sc/datasets/train_data.parquet', tokenizer_name_or_path='unsloth/Mistral-Nemo-Base-2407-bnb-4bit', sequence_length=4096, batch_size=1, fused_optimizer=False, learning_rate=5e-05, lr_warmup_steps=100, training_steps=100, logging_frequency=5, profile=False, profile_step_start=10, profile_step_end=12, grad_max_norm=1, model_dtype='bf16', compile=False, scaling_factor=10, scaling_strategy=<ScalingStrategy.ALL: 'all'>, set_seed=42)
2025-05-22 16:10:07,423 - root - INFO - Setting up DataLoaders...
2025-05-22 16:10:11,199 - root - INFO - Setting seed to 42
2025-05-22 16:10:11,200 - root - INFO - Experiment args: Namespace(dataset='/capstor/store/cscs/ethz/large-sc/datasets/train_data.parquet', tokenizer_name_or_path='unsloth/Mistral-Nemo-Base-2407-bnb-4bit', sequence_length=4096, batch_size=1, fused_optimizer=False, learning_rate=5e-05, lr_warmup_steps=100, training_steps=100, logging_frequency=5, profile=False, profile_step_start=10, profile_step_end=12, grad_max_norm=1, model_dtype='bf16', compile=False, scaling_factor=10, scaling_strategy=<ScalingStrategy.ALL: 'all'>, set_seed=42)
2025-05-22 16:10:11,200 - root - INFO - Setting up DataLoaders...
2025-05-22 16:10:15,409 - root - INFO - Setting up Model...
2025-05-22 16:10:15,410 - root - INFO - Loading a model with scale=10, scaling_strategy=ScalingStrategy.ALL, config:
TransformerModelArgs(dim=2560, n_layers=80, n_heads=80, n_kv_heads=8, multiple_of=256, ffn_dim_multiplier=1.3, norm_eps=1e-05, rope_theta=500000, norm_type='rmsnorm', seq_len=4096, vocab_size=131072)
2025-05-22 16:10:15,410 - root - INFO - Setting up Model...
2025-05-22 16:10:15,410 - root - INFO - Loading a model with scale=10, scaling_strategy=ScalingStrategy.ALL, config:
TransformerModelArgs(dim=2560, n_layers=80, n_heads=80, n_kv_heads=8, multiple_of=256, ffn_dim_multiplier=1.3, norm_eps=1e-05, rope_theta=500000, norm_type='rmsnorm', seq_len=4096, vocab_size=131072)
2025-05-22 16:10:47,628 - root - INFO - Total model parameters: 7,329,958,400
2025-05-22 16:10:47,632 - root - INFO - Starting training!
2025-05-22 16:10:48,405 - root - INFO - Total model parameters: 7,329,958,400
2025-05-22 16:10:48,409 - root - INFO - Starting training!
2025-05-22 16:10:49,584 - root - INFO - Step: 1 | Loss: 11.93 | Tokens per second: 2112.90 | Training tokens per second (%): 19.38 | MFU (%): 11.12 | TFLOPs: 109.94
Traceback (most recent call last):
  File "/iopsstor/scratch/cscs/baraq/lai-proj/train.py", line 119, in <module>
    train(args)
  File "/iopsstor/scratch/cscs/baraq/lai-proj/train.py", line 81, in train
    loss = torch.nn.functional.cross_entropy(logits.flatten(0, 1).float(), labels.flatten(0, 1), reduction="sum")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/torch/nn/functional.py", line 3489, in cross_entropy
    return torch._C._nn.cross_entropy_loss(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.00 GiB. GPU 0 has a total capacity of 94.50 GiB of which 2.68 GiB is free. Including non-PyTorch memory, this process has 90.24 GiB memory in use. Of the allocated memory 85.00 GiB is allocated by PyTorch, and 4.48 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
2025-05-22 16:10:50,365 - root - INFO - Step: 1 | Loss: 11.93 | Tokens per second: 2113.86 | Training tokens per second (%): 19.38 | MFU (%): 11.12 | TFLOPs: 109.99
Traceback (most recent call last):
  File "/iopsstor/scratch/cscs/baraq/lai-proj/train.py", line 119, in <module>
    train(args)
  File "/iopsstor/scratch/cscs/baraq/lai-proj/train.py", line 81, in train
    loss = torch.nn.functional.cross_entropy(logits.flatten(0, 1).float(), labels.flatten(0, 1), reduction="sum")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/torch/nn/functional.py", line 3489, in cross_entropy
    return torch._C._nn.cross_entropy_loss(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.00 GiB. GPU 0 has a total capacity of 94.50 GiB of which 2.85 GiB is free. Including non-PyTorch memory, this process has 90.17 GiB memory in use. Of the allocated memory 85.00 GiB is allocated by PyTorch, and 4.41 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
srun: error: nid006448: task 0: Exited with exit code 1
srun: Terminating StepId=458107.0
srun: error: nid006457: task 1: Terminated
srun: Force Terminated StepId=458107.0
END TIME: Thu May 22 16:10:52 CEST 2025
